99.wc_object/file_stat.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require_relative './wc'
+
+class FileStat
+  def self.display(argv)
+    new(argv).display
+  end
+
+  def initialize(argv)
+    @options, @filenames = extract_options(argv)
+  end
+
+  def display
+    fetch_wc_files.each do |wc_file|
+      text = ''
+      text += wc_file.line_count.to_s.rjust(8) if @options[:l] || @options.empty?
+      text += wc_file.word_count.to_s.rjust(8) if @options[:w] || @options.empty?
+      text += wc_file.byte_count.to_s.rjust(8) if @options[:c] || @options.empty?
+      puts "#{text} #{wc_file.name}"
+    end
+  end
+
+  private
+
+  def extract_options(argv)
+    options = {}
+    opt = OptionParser.new
+    opt.on('-l') { |v| options[:l] = v }
+    opt.on('-w') { |v| options[:w] = v }
+    opt.on('-c') { |v| options[:c] = v }
+    filenames = opt.parse(argv)
+
+    [options, filenames]
+  end
+
+  def fetch_wc_files
+    if @filenames.empty?
+      [Wc.new(text: $stdin.read)]
+    else
+      files = @filenames.map { |filename| File.open(filename) }
+      wc_files = build_wc_files(files)
```

Comment:
42行目と50行まで2回ループするのが非効率です。

```suggestion
      wc_files = @filenames.map do |filename|
        # ... 
      end
```

build_wc_filesはなくして、ここで一気にオブジェクトを作る形でも良いかと思います

Reply:
修正しました！